### ISSUE TITLE
Ensure README & docs use good patterns for prototype vs. instance

### ISSUE THREAD

**Issue by mike-north** [typed-ember/ember-cli-typescript#224]
In [this discussion over at ember-decorators](https://github.com/ember-decorators/ember-decorators/issues/249#issuecomment-401554625), we identified that there are some patterns around service injection in this project's README that are causing errors in Ember 3.1. 

In this piece of code
```ts
// THIS IS A BAD EXAMPLE. DO NOT COPY/PASTE INTO YOUR CODEBASE
import Component from '@ember/component';
import Computed from '@ember/object/computed';
import { inject as service } from '@ember/service';

import Session from 'my-app/services/session';

export default class UserProfile extends Component {
  session: Computed<Session> = service();
}
// THIS IS A BAD EXAMPLE. DO NOT COPY/PASTE INTO YOUR CODEBASE
```
Every instance will get their own service injection. I'm told (thx @stefanpenner) that there's some per-instance cost with the CP anyway (having to do with private ember object internals) but this adds even more per-instance cost -- and a totally avoidable cost at that. 

The correct patten we should be encouraging is

```ts
import Component from '@ember/component';
import Computed from '@ember/object/computed';
import { inject as service } from '@ember/service';

import Session from 'my-app/services/session';

export default class UserProfile extends Component.extend({
   // Injected things like services belong on the prototype!
   session: service('session') // a Computed<Session>
}) {

}
```

This kind of thing should apply to the actions block as well. The current version of the README does things like

```ts
// THIS IS A BAD EXAMPLE. DO NOT COPY/PASTE INTO YOUR CODEBASE
import Component from '@ember/component';

export default class UserProfile extends Component {
  actions = { }
}
// THIS IS A BAD EXAMPLE. DO NOT COPY/PASTE INTO YOUR CODEBASE
```

which will result in each component instance getting its own distinct actions block. From the "ember w/ typescript" apps I've seen while supporting teams, these kinds of mistakes are widespread, and *often made in an effort to help the type system understand the "correct" structure of the ember object*.

We should make sure we're not encouraging developers to unknowingly opt-out of using prototypes to share functions, CP definitions, actions blocks, etc... Lots of folks in our community are just starting to learn about ES6 classes, TypeScript class instance fields and parameter properties, and they don't know any better.

**Comment by chriskrycho** [comment]
Thank you for this detailed writeup! I haven't been back to update the docs since 3.1 came out, but answer this all the time in Slack. (The last few months have afforded me almost no time to work on this, but happily that should be changing this week.)

**Comment by chriskrycho** [comment]
@mike-north if you can write an update to the README, I'll merge it post-haste.

**Comment by mike-north** [comment]
I'll add it to my OSS backlog, but this would be awesome for a new contributor to take a stab at.

@chriskrycho you also may want to consider adding a disclaimer or something to your series of blog articles https://www.chriskrycho.com/2018/typing-your-ember-update-part-2.html (see `3. Performance changes`)

You present the "CP on the instance" vs "CP on the prototype" as a performance optimization, but in Ember 3.1+ one will break (it was never really supported in the first place -- just happened to work) and the other will not.

**Comment by chriskrycho** [comment]
Yes, that is already officially on my own writing backlog! 😆

**Comment by BryanCrotaz** [comment]
I'm on the PR trail :) I lost a few days last week on this so I know the pain everyone else is facing. Now I understand it, I know where the docs are missing the detail.

**Comment by BryanCrotaz** [comment]
@chriskrycho your articles are where I got the wrong instructions from (I first used it in Ember 2.18 and it worked and then my upgrade to Ember 3.1 to fix something else dropped me in a world of pain!) so +1 to @mike-north 's comment.

**Comment by mike-north** [comment]
Resolved by #225
